data-model-layer/data-model.md

```diff
@@ -85,9 +85,42 @@ Boolean is a scalar kind.  Its cardinality is two -- either the value 'true' or
 
 #### Integer kind
 
+Integer is a scalar kind and refers to whole numbers without a fractional
+component.
+
+It is important to consider codec and language limitations that may be imposed
+on the serialization of integers from the IPLD Data Model. For example:
+
+* Some codecs, such as DAG-CBOR, will assume that integers must be within the
+  64-bit signed range and reject anything larger.
+* IPLD libraries, such as go-ipld-prime, limit their in-memory representation
+  of the integer kind to the signed 64-bit range.
+* JavaScript has difficulties safely handling and representing integers outside
+  of the 53-bit unsigned range and differentiating between integers and floats
+	where there is no fractional component.
 
 #### Float kind
 
+Float is a scalar kind and refers to a number with a fractional component,
+represented with a decimal point. This may also include numbers where the
+fractional component is zero (although the distinction between these numbers
+and integers is blurred in some cases—specifically the JavaScript memory model).
+
+While Float is a formal kind in the IPLD Data Model, **it is recommended that
+Float values be avoided when developing systems on IPLD** (and
+content-addressable systems in general) due to the broad scope for introducing
+variability in byte representations. Alternative methods for encoding
+fractional numbers with greater precision and less variability should be
+considered where possible.
```

Comment:
do we want to give some examples here? for example, off the top of my head it's not obvious how I would store a "big int" or a "big float" with arbitrary precision in IPLD. A base-10 string? Some sort of binary encoding in bytes?

Reply:
I've added some examples but would appreciate some sanity checking on them!